File: app/src/main/java/com/damhoe/skatscores/game/application/ports/in/AddScoreToGameUseCase.java
package com.damhoe.skatscores.game.application.ports.in;

import com.damhoe.skatscores.base.Result;
import com.damhoe.skatscores.game.domain.skat.SkatGame;
import com.damhoe.skatscores.game.domain.score.SkatScore;

public interface AddScoreToGameUseCase {
 void addScoreToGame(SkatGame skatGame, SkatScore score);
 Result<SkatScore> removeLastScore(SkatGame skatGame);
}

File: app/src/main/java/com/damhoe/skatscores/game/domain/skat/application/ports/AddScoreToGameUseCase.kt
package com.damhoe.skatscores.game.domain.skat.application.ports

import com.damhoe.skatscores.base.Result
import com.damhoe.skatscores.game.domain.score.SkatScore
import com.damhoe.skatscores.game.domain.skat.SkatGame

interface AddScoreToGameUseCase
{
    fun addScoreToGame(
        skatGame: SkatGame?,
        score: SkatScore?)
    
    fun removeLastScore(skatGame: SkatGame?): Result<SkatScore?>?
}
